OrbitGl/PickingManager.cpp

```diff
@@ -7,21 +7,16 @@
 #include "OpenGl.h"
 #include "OrbitBase/Logging.h"
 
-PickingId PickingManager::GetOrCreatePickableId(std::weak_ptr<Pickable> pickable,
+PickingId PickingManager::GetOrCreatePickableId(std::shared_ptr<Pickable> pickable,
                                                 BatcherId batcher_id) {
-  auto locked_pickable = pickable.lock();
-  CHECK(locked_pickable != nullptr);
-
   absl::MutexLock lock(&mutex_);
   uint32_t pickable_id = 0;
 
-  auto it = pickable_pid_map_.find(locked_pickable.get());
-  if (it != pickable_pid_map_.end()) {
-    pickable_id = it->second;
-  } else {
+  auto it = pickable_pid_map_.find(pickable.get());
+  if (it == pickable_pid_map_.end()) {
     pickable_id = ++pickable_id_counter_;
     pid_pickable_map_[pickable_id] = pickable;
-    pickable_pid_map_[locked_pickable.get()] = pickable_id;
+    pickable_pid_map_[pickable.get()] = pickable_id;
   }
 
   PickingId id = PickingId::Create(PickingType::kPickable, pickable_id_counter_, batcher_id);
```

Comment:
I am beginning to suspect this should use pickable_id here instead of the counter

Reply:
yep - this is a bug - added a test for it.

Reply:
Right, good catch, thanks - I think it only worked by accident, because we are adding all primitives of a Pickable at once...